### PR TITLE
Year last competed shown on team pages

### PR DESCRIPTION
**Pull request by maherb**
The last year a team competed is shown on team pages directly below the rookie year listing EXCEPT when that year is the current year and the team page is the current year, as requested in #1546.

image

### CODE REVIEW

File: models/team.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 from google.appengine.ext import ndb
 from helpers.champ_split_helper import ChampSplitHelper
```
Comment (fangeugene) [inline models/team.py]: Can you revert the whitespace changes made to this file?

File: renderers/team_renderer.py [hunk 2]
```diff
@@ -117,6 +117,14 @@ def render_team_details(cls, handler, team, year, is_canonical):
             district_type = DistrictType.abbrevs[district_abbrev]
             district_name = DistrictType.type_names[district_type]
 
+        last_competed = None
+        participation_query = team_query.TeamParticipationQuery(team.key_name)._query_async()
```
Comment (fangeugene) [inline renderers/team_renderer.py]: This should be `participation_future = team_query.TeamParticipationQuery(team.key_name).fetch_async()` and put on line 31 next to all the other queries.

File: renderers/team_renderer.py [hunk 3]
```diff
@@ -117,6 +117,14 @@ def render_team_details(cls, handler, team, year, is_canonical):
...
+        participation_years = participation_query.get_result()
+        print participation_years
```
Comment (fangeugene) [inline renderers/team_renderer.py]: Stray print

File: renderers/team_renderer.py [hunk 4]
```diff
@@ -28,6 +28,7 @@ def render_team_details(cls, handler, team, year, is_canonical):
         social_media_future = media_query.TeamSocialMediaQuery(team.key.id()).fetch_async()
         robot_future = Robot.get_by_id_async('{}_{}'.format(team.key.id(), year))
         team_districts_future = team_query.TeamDistrictsQuery(team.key.id()).fetch_async()
+        participation_future = team_query.TeamParticipationQuery(team.key.id())._query_async()
```
Comment (fangeugene) [inline renderers/team_renderer.py]: this should be `fetch_async()`

**Comment by fangeugene** [comment]
The whitespace doesn't seem to be fixed. Is your text editor changing things?

**Comment by maherb** [comment]
At one point in figuring out how to do it, I modified the team model. I got rid of my changes, but had messed with the whitespace in the process. I'm not super git savvy, what command can I use to just revert the whitespace?

**Comment by maherb** [comment]
Whitespace is fixed.

File: renderers/team_renderer.py (hunk 4 above)
Comment (phil-lopreiato) [inline renderers/team_renderer.py]: `_query_async` is some internal ndb method, you should stick to the documented interface, which you can see here: https://cloud.google.com/appengine/docs/python/ndb/queryclass

Like Eugene said, `fetch_async` is what you want here

**Comment by phil-lopreiato** [comment]
ping @maherb on the comment above

File: templates_jinja2/team_partials/team_info.html [hunk 5]
```diff
@@ -15,6 +15,7 @@
       {% if team.location %}<span class="glyphicon glyphicon-map-marker"></span> From <a href="http://maps.google.com/maps?q={{ team.location|urlencode }}" target="_blank">{{ team.location }}</a><br>{% endif %}
       {% if team.name %}<span class="glyphicon glyphicon-info-sign"></span> aka <em>{{ team.name }}</em><br>{% endif %}
       {% if team.rookie_year %}<span class="glyphicon glyphicon-calendar"></span> First competed in {{ team.rookie_year }}<br>{% endif %}
+      {% if last_competed and year != current_year %}Last competed in {{ last_competed }}.{% endif %}
```
Comment (SPCOxion) [inline templates_jinja2/team_partials/team_info.html]: Missing a \<br\> tag. This causes the "last competed in" stat and the team website to be on the same line.
There's also a period that looks out of place.
2016-08-06 05_38_33-d penguineers - team 1717 2015 - the blue alliance

Comment (SPCOxion) [inline templates_jinja2/team_partials/team_info.html]: Missing a \<br\> tag.
2016-08-06 05_38_33-d penguineers - team 1717 2015 - the blue alliance

Comment (fangeugene) [inline templates_jinja2/team_partials/team_info.html]: Fixed in https://github.com/the-blue-alliance/the-blue-alliance/commit/5524b71c589a2ac2ab95b5ff75c2d3385fec69b6

**Pull request merged**